### PR TITLE
Convert some tests to property tests

### PR DESCRIPTION
**Pull request by 00-kat**
This PR adds hypothesis and changes some tests that have random parametrized literals to property tests.

Maybe I'll add more tests that use hypothesis, maybe not; I haven't yet looked around to see if anything untested can be property tested.

I have no idea if this is considered useful in its current state.

### CODE REVIEW

File: tests/test_utils.py [hunk 1]
```diff
@@ -144,20 +121,7 @@ def test_post_is_not_solved(names: list[str]) -> None:
...
-        [False, True, True],
-        [{"a": 12}, {"b": 14, "c": 23}, {"d": 183, "e": 21}],
-        [..., ..., ...],
-        [None, None],
-        [[1, 2, 3], [2, 3, 4], [3, 4, 5], [4, 5, 6], [5, 6, 7]],
-        [(1, 2, 3), (2, 3, 4), (3, 4, 5), (4, 5, 6), (5, 6, 7)],
-    ],
-)
-@pytest.mark.parametrize("start", [0, 1, 2, 3, 4, 5, -1, -2, -3, -4, -5, 102, -99, 41])
+@given(st.lists(st.from_type(type)), st.integers())
```
Comment (00-kat) [inline tests/test_utils.py:124]: In case you're wondering what this is, `from_type(type)` draws from everything that `from_type()` can generate.

File: tests/test_message_moving.py [hunk 2]
```diff
@@ -88,25 +90,14 @@ def test_message_can_be_moved(type_: dc.MessageType, result: bool) -> None:
...
-        "terminal emulator",
-    ],
-)
+@given(st.text())
 def test_unattachable_embed(elem: str) -> None:
     # Require the (escaped) string to be in the returned embed. repr() on a string wraps
     # it in quotes, so remove those too.
-    assert repr(elem)[1:-1] in repr(_unattachable_embed(elem).to_dict()).casefold()
+    assert (
+        repr(elem)[1:-1].casefold()
```
Comment (00-kat) [inline tests/test_message_moving.py:98]: The casefold is done after the repr instead of in a [`.map()`] because otherwise `\Uxxxxxxxx` literals fail (they're `\uxxxxxxxx` on the rhs but not the lhs).

[`.map()`]: https://hypothesis.readthedocs.io/en/latest/reference/strategies.html#hypothesis.strategies.SearchStrategy.map

File: tests/test_utils.py [hunk 3]
```diff
@@ -41,21 +43,7 @@ def test_is_mod() -> None:
...
-        -1,
-        0,
-        1274687918204,
-        19267480912409,
-        2**200,
-        9999999999999999999999999,
-        1756274809124124,
-    ],
-)
+@given(st.integers())
```
[the remark this answers is not in the document]
Comment (00-kat) [reply tests/test_utils.py:46]: Oh I didn't realize I forgot to filter out the mod role id, oops.

File: tests/test_utils.py [hunk 4]
```diff
@@ -259,11 +221,9 @@ async def edit(**kwargs: Any) -> None:
     assert suppressed
 
 
-@pytest.mark.parametrize(
-    ("additions", "deletions", "changed_files"),
-    [(1, 2, 3), (2, 3, 4), (-12, 0, 0), (-43, -21, -17), (0, 0, 1)],
-)
+@given(st.integers(), st.integers(), st.integers())
 def test_format_diff_note(additions: int, deletions: int, changed_files: int) -> None:
+    assume(additions and deletions and changed_files)
```
[the remark this answers is not in the document]
Comment (00-kat) [reply tests/test_utils.py:226]: Oh actually yeah the `format_diff_note` one has
```py
if not (additions or deletions or changed_files):
```
which is basically the same as yours (other than the `changed_files` edge case which I'll update).

I'll change this assume to use `or` instead of `and` too ^^.

**Review by CoderJoshDK — APPROVED** [review]
Assuming that trag1c wants to introduce `hypothesis`, this looks good to me.
